peachjam/views/documents.py

```diff
@@ -9,6 +11,55 @@
 from peachjam.registry import registry
 
 
+class RedirectResolver:
+
+    RESOLVER_MAPPINGS = {
+        "africanlii": {"code": "aa", "domain": "agp.africanlii.org"},
```

Comment:
`code` is ambiguous. In FRBR URI terms, we have three:

1. country code (eg. `tz`)
2. locality code (eg. `znz`) which must be combined with a country code to be unique
3. place code -- the combination of the two (eg. `tz-znz`)

We need to be able to juggle those. For example, LawLibrary is authoritative for all South African content, so it covers `za` and all localities (`za-*`). However, TanzLII is authoritative for `tz` and `tz-*` except `tz-znz`.

Matching on a full place code  should be considered a "tighter" match than matching on just the country code. So here's a suggestion that will make it easier to handler the edge cases like ZanzibarLII:

1. change `code` to `country_code` for all of these settings, except ZanzibarLII, where change it to `place_code`
2. first check for an exact match on `place_code`; if found, use that authority
3. then check for a match on `country_code`; if found, use that authority (this will be the common case)
4. nothing matches, so give up